src/runtime/volatile/register_nxpmk66f18.go

```diff
@@ -0,0 +1,79 @@
+// +build nxp,mk66f18
+
+package volatile
+
+import "unsafe"
+
+const registerBase = 0x40000000
+const registerEnd = 0x40100000
+const bitbandBase = 0x42000000
+const ptrBytes = unsafe.Sizeof(uintptr(0))
+
+//go:inline
+func bitbandAddress(reg uintptr, bit uint8) uintptr {
+	if uintptr(bit) > ptrBytes*8 {
+		panic("invalid bit position")
+	}
+	if reg < registerBase || reg >= registerEnd {
+		panic("register is out of range")
+	}
+	return (reg-registerBase)*ptrBytes*8 + uintptr(bit)*ptrBytes + bitbandBase
+}
+
+// Special types that causes loads/stores to be volatile (necessary for
+// memory-mapped registers).
+type BitRegister struct {
+	Reg uint32
```

Comment:
Because the K66 docs describe these as 32-bit registers. Endianness is something I try to avoid thinking about. Would treating an address of a 32-bit register as an 8-bit value safely allow setting the low bit?